lib/reqTemplate.js

```diff
@@ -159,6 +162,21 @@ function _createTemplateResolvers(origSpec, subspec, reqPart, path) {
     }
 }
 
+function calculateHash(context) {
+    // Need to remove x-request-id header if it's present
+    var reqId = context.request.headers && context.request.headers['x-request-id'];
+    if (reqId) {
+        delete context.request.headers['x-request-id'];
+    }
+
+    var result = sha1(stringify(context.request));
```

Comment:
We should ensure consistent hashing. That means checking for standard request components (uri, headers, body, query) so as to avoid different hashes in situations where some are missing. E.g., these two requests will yield different hashes even though they should be treated equally:

```yaml
request1:
  uri: /a/simple/path
  headers: {}
  body: ''

request2:
  uri: /a/simple/path
  body: {}
```